File: prolog/space-age/space_age.pl
space_age(Planet, AgeSec, Years) :-
    orbital_period("Earth", OrbitalPeriod),
    factor(Planet, Factor),
    Years is AgeSec / (OrbitalPeriod * Factor).

orbital_period("Earth", 31557600).

factor("Mercury", 0.2408467).
factor("Venus", 0.61519726).
factor("Earth", 1).
factor("Mars", 1.8808158).
factor("Jupiter", 11.862615).
factor("Saturn", 29.447498).
factor("Uranus", 84.016846).
factor("Neptune", 164.79132).

File: erlang/space-age/src/space_age.erl
-module(space_age).

-export([age/2]).

-type planet() :: mercury
                | venus
                | earth
                | mars
                | jupiter
                | saturn
                | uranus
                | neptune.

-type second() :: pos_integer().

-spec age(planet(), second()) -> float().
age(mercury, Seconds) -> calc(0.2408467, Seconds);
age(venus, Seconds) -> calc(0.61519726, Seconds);
age(earth, Seconds) -> calc(1.0, Seconds);
age(mars, Seconds) -> calc(1.8808158, Seconds);
age(jupiter, Seconds) -> calc(11.862615, Seconds);
age(saturn, Seconds) -> calc(29.447498, Seconds);
age(uranus, Seconds) -> calc(84.016846, Seconds);
age(neptune, Seconds) -> calc(164.79132, Seconds).

-spec calc(float(), second()) -> float().
calc(Factor, Seconds) -> Seconds / (Factor * 31557600).